src/openrct2/drawing/X8DrawingEngine.cpp

```diff
@@ -188,9 +188,11 @@ void X8DrawingEngine::BeginDraw()
     if (!IntroIsPlaying())
     {
         // HACK we need to re-configure the bits if light fx has been enabled / disabled
-        if (_lastLightFXenabled != (Config::Get().general.EnableLightFx != 0))
+        if (_lastLightFXenabled != Config::Get().general.EnableLightFx)
         {
             Resize(_width, _height);
+            GfxInvalidateScreen();
```

Comment:
I had initially put this call in the Options window, but `Resize` negates its effects. Moving it here had the desired effect.